passportAuth.go

```diff
@@ -47,25 +46,20 @@ func (p *PassportAuth) Authorize(c *http.Client, rq *http.Request, path string)
 
 // Verify verifies if the authentication is good
 func (p *PassportAuth) Verify(c *http.Client, rs *http.Response, path string) (redo bool, err error) {
-	if rs.StatusCode == 302 {
-		if rs.Header.Get("WWW-Authenticate") != "" {
+	switch rs.StatusCode {
+	case 301, 302, 307, 308:
+		redo = true
+		if rs.Header.Get("Www-Authenticate") != "" {
 			// re-authentication required as we are redirected to the login page
-			err = p.genCookies(&rs.Header)
-			if err != nil {
-				return
-			}
-			redo = true
+			p.cookieMutex.Lock()
```

Comment:
we do not need the mutex all. it is handled outside. the authenticator is dedicated to the request.

Reply:
Quick sanity question: The authenticator is cloned for each request and not a new one created?

Reply:
for each operation. in the best case its one request.

Reply:
> I just noticed something. The `p.inhibitRedirect` approach will not work when the authenticator is used concurrently. So I think the ~easiest~ cleanest solution would be to simply go back to the `Verify` function resolving the redirect.

each operation like `GET /foo.txt` get`s i an own authenticator instance.
the instance is bound to origin request and all roundtrips within one go-routine.
this should work.
could you add some tests?

Reply:
the auth_test.go contains a `TestRedirectAuthWIP` which could be used as a starting point to verify the auth flow.

Reply:
This is not going to work with the preemptive auth!

Reply:
> This is not going to work as preemptive auth!

Could you elaborate on this? Is this still a problem with the latest changes? What exactly is the problem?